src/components/cardbuilder/cardBuilderUtils.ts

```diff
@@ -0,0 +1,356 @@
+import { randomInt } from '../../utils/number';
+
+const ASPECT_RATIOS = {
+    portrait: (2 / 3),
+    backdrop: (16 / 9),
+    square: 1,
+    banner: (1000 / 185)
+};
+
+/**
+ * Determines if the item is live TV.
+ * @param {string} itemType - Item type to use for the check.
+ * @returns {boolean} Flag showing if the item is live TV.
+ */
+export const isUsingLiveTvNaming = (itemType: string): boolean => itemType === 'Program' || itemType === 'Timer' || itemType === 'Recording';
+
+/**
+ * Resolves Card action to display
+ * @param opts options to determine the action to return
+ */
+export const resolveAction = (opts: { defaultAction: string, isFolder: boolean, isPhoto: boolean }): string => {
+    if (opts.defaultAction === 'play' && opts.isFolder) {
+        // If this hard-coding is ever removed make sure to test nested photo albums
+        return 'link';
+    } else if (opts.isPhoto) {
+        return 'play';
+    } else {
+        return opts.defaultAction;
+    }
+};
+
+/**
+ * Checks if the window is resizable.
+ * @param {number} windowWidth - Width of the device's screen.
+ * @returns {boolean} - Result of the check.
+ */
+export const isResizable = (windowWidth: number): boolean => {
+    const screen = window.screen;
+    if (screen) {
+        const screenWidth = screen.availWidth;
+
+        if ((screenWidth - windowWidth) > 20) {
+            return true;
+        }
+    }
+
+    return false;
+};
+
+/**
+ * Resolves mixed shape based on aspect ratio
+ * @param primaryImageAspectRatio image aspect ratio that determines mixed shape
+ */
+export const resolveMixedShapeByAspectRatio = (primaryImageAspectRatio: number | null | undefined) => {
+    if (primaryImageAspectRatio === undefined || primaryImageAspectRatio === null) {
+        return 'mixedSquare';
+    }
+
+    if (primaryImageAspectRatio >= 1.33) {
+        return 'mixedBackdrop';
+    } else if (primaryImageAspectRatio > 0.71) {
+        return 'mixedSquare';
+    } else {
+        return 'mixedPortrait';
+    }
+};
+
+type CardCssClassOpts = {
+    shape?: string,
+    cardCssClass?: string,
+    cardClass?: string,
+    tagName?: string,
+    itemType: string,
+    childCount?: number,
+    showChildCountIndicator: boolean,
+    isTV: boolean,
+    enableFocusTransform: boolean,
+    isDesktop: boolean
+};
+
+/**
+ * Resolves applicable Card CSS classes
+ * @param opts options for determining which CSS classes are applicable
+ */
+export const resolveCardCssClasses = (opts: CardCssClassOpts): string => {
+    const classNames = ['card'];
```

Comment:
You could use the `classnames` utility for this. https://github.com/JedWatson/classnames

Reply:
Awesome, I didn't quite realize this dependency was already in the project. Will optimize these functions then. Thanks!

Reply:
Updated to use `classnames`. Great catch!